File: src/main/java/com/rtomyj/skc/exception/YgoException.java
package com.rtomyj.skc.exception;

import com.rtomyj.skc.enums.ErrorType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import org.springframework.http.HttpStatus;


@Data
@EqualsAndHashCode(callSuper = false)
@Builder
@AllArgsConstructor
public class YgoException extends RuntimeException
{
	private final String message;
	private final HttpStatus httpStatus;
	private final ErrorType errorType;
}

File: src/main/java/com/rtomyj/skc/exception/YgoException.kt
package com.rtomyj.skc.exception

import lombok.Builder
import lombok.Data
import lombok.EqualsAndHashCode
import java.lang.RuntimeException

@Data
@EqualsAndHashCode(callSuper = false)
@Builder
class YgoException(override val message: String, val errorType: ErrorType)
	: RuntimeException()